incubating/vault-next-gen/script/vault.py

```diff
@@ -0,0 +1,205 @@
+import hvac
+import os
+import sys
+import json
+import base64
+
+from step_utility import StepUtility
+
+class Secret:
+    def __init__(self, export_name, path, secret_name, secret_value):
+        self.export_name = export_name
+        self.path = path
+        self.secret_name = secret_name
+        self.secret_value = secret_value
+
+class Environment:
+    def __init__(self, vault_addr, vault_auth_method, vault_token,
+        approle_role_id, approle_secret_id,
+        vault_client_cert_base64, vault_client_key_base64,
+        mount_point, vault_kv_version, verbose):
+        self.vault_addr = vault_addr
+        self.vault_auth_method = vault_auth_method
+        self.vault_token = vault_token
+        self.approle_role_id = approle_role_id
+        self.approle_secret_id = approle_secret_id
+        self.vault_client_cert_base64 = vault_client_cert_base64
+        self.vault_client_key_base64 = vault_client_key_base64
+        self.mount_point = mount_point
+        self.vault_kv_version = vault_kv_version
+        self.verbose = verbose
+
+
+def main():
+    env, current_environment = environment_setup()
+    secrets, path_set = secrets_setup(env)
+    client = vault_authentication(current_environment)
+    retrieved_secrets = get_secrets(client, current_environment, secrets, path_set)
+    export_secrets(retrieved_secrets)
+
+
+def environment_setup():
+    # Grab all of the environment variables
+    env = os.environ
+    vault_addr = StepUtility.getEnvironmentVariable('VAULT_ADDR', env)
```

Comment:
Recommendation: `vault_addr = os.getenv('VAULT_ADDR')`

Reply:
I broke out the code so that I could create a portable "step utility" class where I can start putting useful methods over time for step processing